### PR TITLE
Stability Fixes

### PR DESCRIPTION
**Pull request by h0ngcha0**
- Reduce re-rendering & API calls
- Trim transaction history
- Enable sweep when ALPH only
- Discover active accounts for new users

### CODE REVIEW

File: packages/extension/src/background/accountMessaging.ts [hunk 1]
```diff
@@ -29,6 +29,25 @@ export const handleAccountMessage: HandleMessage<AccountMessage> = async ({
       })
     }
 
+    case "DISCOVER_ACCOUNTS": {
+      const { networkId } = msg.data
+      try {
```
[the remark this answers is not in the document]
Comment (h0ngcha0) [reply packages/extension/src/background/accountMessaging.ts:34]: e.g. when API calls to explorer backend fail in the background

File: packages/extension/src/shared/transactions/index.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      const transactions = await getTransactions(account)
+      transactionsPerAccount.set(account, transactions)
+    }),
+  )
+
+  return transactionsPerAccount
+}
+
+// The number of transactions fetched has the following constraints:
+// 1) At most 50 transactions (3.1kb per tx, 50 txs = 155kb per account),
```
[the remark this answers is not in the document]
Comment (h0ngcha0) [reply packages/extension/src/shared/transactions/index.ts:114]: Yeah, we will still display them if it's more than 50tx in the last two days.
Diomark is a heavy user and he has > 1000 txs over a pretty long time, maybe this limit is reasonable, WDYT?

[the remark this answers is not in the document]
Comment (Lbqds) [reply packages/extension/src/shared/transactions/index.ts:114]: Thanks for the explanation, sounds good to me 👍

**Review by Lbqds — APPROVED** [review]
LGTM 👍

**Comment by polarker** [comment]
Nice. I will go for a quick review later today.

**Comment by h0ngcha0** [comment]
If there are no accounts for a network, there is now a button to discover active accounts

image

**Pull request merged**